spyder/plugins/__init__.py

```diff
@@ -591,3 +597,25 @@ def get_plugin_actions(self):
     def register_plugin(self):
         """Register plugin in Spyder's main window"""
         raise NotImplementedError
+
+    def check_compatibility(self):
+        """
+        This method can be implemented to check compatibility of plugin on
+        a given os.

```

Comment:
Better instead

    ... to check compatibility of the plugin for a given condition.